diff -ru papi-6.0.0.1.orig/src/configure papi-6.0.0.1/src/configure
--- papi-6.0.0.1.orig/src/configure	2020-04-03 03:07:00.000000000 -0500
+++ papi-6.0.0.1/src/configure	2020-07-12 15:38:12.346963979 -0500
@@ -3374,6 +3374,20 @@
 	CC_COMMON_NAME="unknown"
 fi
 
+# Lets figure out what F77 actually is...
+# Used in later checks to set compiler specific options
+if `$F77 -V 2>&1 | grep '^Intel(R) Fortran' >/dev/null 2>&1` ; then
+	F77_COMMON_NAME="ifort"
+elif `$F77 -v 2>&1 | grep 'gcc version' >/dev/null 2>&1` ; then
+	F77_COMMON_NAME="gfortran"
+elif `$F77 -qversion 2>&1 | grep 'IBM XL Fortran' >/dev/null 2>&1`; then
+	F77_COMMON_NAME="xlf"
+elif `$F77 -V 2>&1 | grep '^pgf77 ' >/dev/null 2>&1`; then
+	F77_COMMON_NAME="pgf77"
+else
+	F77_COMMON_NAME="unknown"
+fi
+
 #prevent icc warnings about overriding optimization settings set by AC_PROG_CC
 # remark #869: parameter was never referenced
 # remark #271: trailing comma is nonstandard
@@ -3384,7 +3398,7 @@
   fi
 fi
 
-if test "$F77" = "ifort" -a "$MIC" = "yes"; then
+if test "$F77_COMMON_NAME" = "ifort" -a "$MIC" = "yes"; then
   F77="$F77 -mmic -fPIC"
 fi
 
@@ -6653,11 +6667,11 @@
 
 
 
-if test "$F77" = "pgf77"; then
+if test "$F77_COMMON_NAME" = "pgf77"; then
   FFLAGS="$FFLAGS -Wall -Mextend"
-elif test "$F77" = "ifort"; then
+elif test "$F77_COMMON_NAME" = "ifort"; then
   FFLAGS="$FFLAGS -warn all"
-elif test "$F77" != "xlf"; then
+elif test "$F77_COMMON_NAME" != "xlf"; then
   FFLAGS="$FFLAGS -ffixed-line-length-132"
 fi
 
