diff -ur papi-7.0.0.orig/src/configure papi-7.0.0/src/configure
--- papi-7.0.0.orig/src/configure	2022-11-14 11:44:31.000000000 -0600
+++ papi-7.0.0/src/configure	2024-02-14 12:18:15.000000000 -0600
@@ -7131,19 +7131,19 @@
   fi
 
   if test "x$comp" = "xsysdetect" ; then
-      if test "x`find $PAPI_CUDA_ROOT -name "cuda.h"`" != "x" ; then
+      if test "x`find ${PAPI_CUDA_ROOT-.} -name "cuda.h"`" != "x" ; then
           CFLAGS="$CFLAGS -DHAVE_CUDA"
       fi
 
-      if test "x`find $PAPI_CUDA_ROOT -name "nvml.h"`" != "x" ; then
+      if test "x`find ${PAPI_CUDA_ROOT-.} -name "nvml.h"`" != "x" ; then
 		  CFLAGS="$CFLAGS -DHAVE_NVML"
 	  fi
 
-      if test "x`find $PAPI_ROCM_ROOT -name "hsa.h"`" != "x" ; then
+      if test "x`find ${PAPI_ROCM_ROOT-.} -name "hsa.h"`" != "x" ; then
           CFLAGS="$CFLAGS -DHAVE_ROCM"
       fi
 
-      if test "x`find $PAPI_ROCM_ROOT -name "rocm_smi.h"`" != "x" ; then
+      if test "x`find ${PAPI_ROCM_ROOT-.} -name "rocm_smi.h"`" != "x" ; then
           CFLAGS="$CFLAGS -DHAVE_ROCM_SMI"
       fi
   fi
